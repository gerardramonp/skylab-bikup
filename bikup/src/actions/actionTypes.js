export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
