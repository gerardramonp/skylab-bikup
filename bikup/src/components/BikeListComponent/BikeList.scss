$light-bg-color: white;
$light-highlight-color: #eeeeee;
$main-color: #0c9;
$dark-highlight-color: #393e46;

.bikelist {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    padding: 32px;
}

.bikelist__top {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    margin-bottom: 24px;
    padding: 0 12px;
}

.bikelist__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
}

.strava__connect-btn {
    height: 40px;
    cursor: pointer;
}

.bikelist__cards {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.bikelist__add {
    width: 95%;
    height: 40px;
    background: none;
    border: 1px solid $main-color;
    border-radius: 5px;
    color: $main-color;
    font-size: 18px;
    cursor: pointer;
}

.bikelist__add:hover {
    color: black;
    background-color: $main-color;
}
