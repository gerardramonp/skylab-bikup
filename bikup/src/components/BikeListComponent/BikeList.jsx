import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import bikeStore from '../../stores/bikeStore';
import { loadUserBikeList } from '../../actions/bikeActions';

import StandardAside from '../StandardAside/StandardAside';
import BikeCard from './BikeCardComponent/BikeCard';

import '../../App.scss';
import './BikeList.scss';

function BikeList(props) {
    const [bikeList, setBikeList] = useState([]);

    const userId = '5f4e6c437edd9832f01d6cc5';
    const bikeId = '5f4e48d27edd9832f01d6cc4';
    const bikeName = 'my Bike Name';

    useEffect(() => {
        bikeStore.addChangeListener(onChange);
        if (bikeList.length === 0) {
            loadUserBikeList(userId);
        }
        return () => bikeStore.removeChangeListener(onChange);
    }, [bikeList.length]);

    function onChange() {
        setBikeList(bikeStore.getBikeList());
    }

    return (
        bikeList && (
            <div className="bikelist">
                <h2>Your Bikes</h2>
                <div className="bikelist__content">
                    <img
                        className="strava__connect-btn"
                        src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f4f63b8021a9d482184baf2/3cca3ad9320164155dfbb9d09ff7982f/btn_strava_connectwith_orange%402x.png"
                        alt="connect with strava"
                    />
                    <div className="bikelist__cards">
                        {bikeList.map((bike) => {
                            return (
                                <BikeCard key={bike.bikeName} bikeInfo={bike} />
                            );
                        })}
                    </div>
                    <button className="bikelist__add">+ Add new bike</button>
                    <div className="bikelist__challenges-giveaways">
                        <StandardAside />
                    </div>
                </div>
            </div>
        )
    );
}

export default BikeList;
