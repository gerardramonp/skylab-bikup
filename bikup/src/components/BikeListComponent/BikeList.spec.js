import React from 'react';
import renderer from 'react-test-renderer';
import BikeList from './BikeList';
import { BrowserRouter as Router } from 'react-router-dom';
import dispatcher from '../../dispatcher';
import actionTypes from '../../actions/actionTypes';

describe('Bike List Component', () => {
	let tree = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<BikeList />
			</Router>
		);
	});

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('Should call onChange function', () => {
		dispatcher.dispatch({
			type: actionTypes.LOAD_BIKE_BY_ID,
			data: 'random',
		});

		expect(tree).toBeTruthy();
	});
});
