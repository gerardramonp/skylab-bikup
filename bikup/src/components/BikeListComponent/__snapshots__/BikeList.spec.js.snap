// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bike List Component should match 1`] = `
Array [
  <div
    className="mobile__menu"
  >
    <div
      className="menu__decorator"
      onClick={[Function]}
    >
      <div
        className="decorator__line"
      />
    </div>
    <ul
      className="menu__list"
    >
      <li
        className="menu__option"
        onClick={[Function]}
      >
        <img
          alt="Logout"
          className="option__img"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5a496d520b7b130a5d4819/7f0d35433cdacef755bec9d18cbcce2c/logout.png"
        />
        <p
          className="option__p"
        >
          Logout
        </p>
      </li>
    </ul>
  </div>,
  <header
    className="header"
  >
    <div
      className="header__mobile"
    >
      <div
        className="mobile__nav"
      >
        <img
          alt="logo"
          className="logo"
          src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
        />
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="ExploreMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png"
          />
        </a>
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="MyBikesMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png"
          />
        </a>
        <a
          aria-current={null}
          href="/user"
          onClick={[Function]}
        >
          <img
            alt="ProfileMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png"
          />
        </a>
        <img
          alt="menu"
          className="nav__menu"
          onClick={[Function]}
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png"
        />
      </div>
    </div>
    <div
      className="header__desktop"
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <p
        className="appName"
      >
        bikUP
      </p>
      <ul
        className="header__navigation"
      >
        <a
          aria-current={null}
          className="navigation__item"
          href="/bikes"
          onClick={[Function]}
        >
          My Bikes
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/search"
          onClick={[Function]}
        >
          Explore
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/user"
          onClick={[Function]}
        >
          Profile
        </a>
      </ul>
      <div
        className="flex-spacer"
      />
      <button
        className="login__button login__button--header"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </header>,
  <div
    className="bikelist"
  >
    <div
      className="bikelist__content"
    >
      <div
        className="bikelist__top"
      >
        <h2>
          Your Bikes
        </h2>
        <a
          aria-current={null}
          className="bikelist__add--desktop desktop"
          href="/bikes/new-bike"
          onClick={[Function]}
        >
          + Add new bike
        </a>
      </div>
      <div
        className="bikelist__cards"
      >
        <div
          className="loadingscreen"
        >
          <img
            alt="loading gif"
            className="loading__giff"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/25b7845da50467b914bf64c612ba27eb/Spinner-1s-200px.gif"
          />
          <p
            className="bikelist__loading"
          >
            Loading bike list...
          </p>
          <p
            className="bikelist__nobikes hidden"
          >
            Oops! Seems that you don't have any bike yet 
            <br />
            <br />
            Try adding a new one
          </p>
        </div>
      </div>
      <a
        aria-current={null}
        className="bikelist__add mobile"
        href="/bikes/new-bike"
        onClick={[Function]}
      >
        + Add new bike
      </a>
    </div>
    <div
      className="bikelist__challenges-giveaways"
    >
      <div
        className="standardaside"
      >
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Challenges
          </h3>
          <div
            className="standardaside__card"
          >
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  BIrlfriend
                   -
                </div>
                <p>
                  Travel 5000km with a bike
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  CompoManiac
                   -
                </div>
                <p>
                  Change the same component 3 times
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
          </div>
        </div>
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Give Away
          </h3>
          <div
            className="standardaside__card"
          >
            <p>
              something free
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
